test/Sentry.Tests/Internals/Http/CachingTransportTests.cs

```diff
@@ -0,0 +1,227 @@
+using System;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using FluentAssertions;
+using NSubstitute;
+using Sentry.Extensibility;
+using Sentry.Internal.Http;
+using Sentry.Protocol.Envelopes;
+using Sentry.Testing;
+using Xunit;
+
+namespace Sentry.Tests.Internals.Http
+{
+    public class CachingTransportTests : IDisposable
+    {
+        private string CacheDirectoryPath { get; } = Path.Combine(
+            Directory.GetCurrentDirectory(),
```

Comment:
This lets us inspect the actual cache directory more easily. Plus I don't like polluting system directories if I have a choice.

Reply:
Note: with the current setup, the cache directory ends up inside `bin/debug` directory in tests.

Reply:
https://github.com/lucas-zimerman/ContribSentryDotNet/blob/master/ContribSentry.Testing/TempFolder.cs

Something like this would be nice for testing the cached folder, it'll delete the files once the test is finished with or without an exception.

Reply:
That does not work if the test is canceled by xUnit timeout, but I think it looks nicer so I'll refactor to that